python/hail/tests/test_expr.py

```diff
@@ -1152,3 +1152,88 @@ def test_interval_ops(self):
         self.assertTrue(li.overlaps(li4).value)
         self.assertFalse(li.overlaps(li3).value)
         self.assertFalse(li.overlaps(li5).value)
+
+    def test_mendel_error_code(self):
+        locus_auto = hl.Locus('2', 20000000)
+        locus_x_par = hl.get_reference('default').par[0].start
+        locus_x_nonpar = hl.Locus(locus_x_par.contig, locus_x_par.position - 1)
+        locus_y_nonpar = hl.Locus('Y', hl.get_reference('default').lengths['Y'] - 1)
+
+        self.assertTrue(hl.all(lambda x: x, hl.array([
+            hl.literal(locus_auto).in_autosome(),
+            hl.literal(locus_auto).in_autosome_or_par(),
+            ~hl.literal(locus_x_par).in_autosome(),
+            hl.literal(locus_x_par).in_autosome_or_par(),
+            ~hl.literal(locus_x_nonpar).in_autosome(),
+            hl.literal(locus_x_nonpar).in_x_nonpar(),
+            ~hl.literal(locus_y_nonpar).in_autosome_or_par(),
+            hl.literal(locus_y_nonpar).in_y_nonpar()
+        ])).value)
+
+        hr = hl.Call([0, 0])
+        het = hl.Call([0, 1])
+        hv  = hl.Call([1, 1])
+        nocall = None
+
+        expected = {
+            (locus_auto, True, hv, hv, het): 1,
+            (locus_auto, False, hv, hv, het): 1,
+            (locus_x_par, True, hv, hv, het): 1,
+            (locus_x_par, False, hv, hv, het): 1,
+            (locus_auto, True, hr, hr, het): 2,
```

Comment:
What is the goal of this test? It's not iterating over every possibility. What happens if sex is missing?

Reply:
oh, that's a good test. I think it should work for autosomes and par, and return missing for the rest.

Reply:
looking over the code, that's what it'll do, but I'll test it.

Reply:
This test tests a bunch of specific cases with expected behavior. I'm happy to improve it if you have suggestions!